doc/developer/platform/ux.md

```diff
@@ -1,20 +1,902 @@
+
+There are no relevant privileges on sinks.
+
+We do not yet know how to provide sinks in Materialize Platform. We are,
+however, aware of at least two important user experience constraints:
+
+  * The default sink experience must be "exactly once." If a cluster restarts,
+    downstream sinks must not reexport data that was previously exported.
+
+  * Sinks must be prepared for the behavior of the COMPUTE layer to change after
+    a version upgrade. If we fix a bug in COMPUTE that changes the data computed
+    by an index, sinks must detect and communicate this behavior change to
+    users. Options include marking the sink as errored and asking users to
+    recreate the sink, or emitting the diff between the old computation and
+    new computation and then proceeding with export of new values.
+
+### Table
+
+The user experience of a table is unchanged.
+
+Users cannot read data from a table on which they do not have the `SELECT`
+privilege.
+
+### View
+
+The user experience of a view is largely unchanged.
+
+The existing `CREATE MATERIALIZED VIEW` statement is extended to support
+specifying the cluster in which to create the index:
+
+```
+CREATE MATERIALIZED VIEW <name> FROM ... [IN CLUSTER <cluster>]
+```
+
+If no cluster is explicitly specified, the active cluster is used. If there
+is no active cluster, the `CREATE VIEW` statement is rejected.
+
+Users cannot read data from a view on which they do not have the `SELECT`
+privilege. Note that if the view references sources or other views, privileges
+on those sources and other views are not considered when determining whether a
+user has access to the view.
+
+### Index
+
+The existing `CREATE INDEX` statement is extended to support specifying the
+cluster in which to create the index:
+
+```
+CREATE INDEX <name> ON <object> (<col>, ...) [IN CLUSTER <cluster>]
+```
+
+If no cluster is explicitly specified, the active cluster is used. If there
+is no active cluster, the `CREATE INDEX` statement is rejected.
+
+`SHOW INDEXES` and `mz_indexes` will gain an additional column specifying to
+which cluster the index belongs.
+
+There are no access control restrictions on individual indexes. Access to
+indexes is controlled by the permissions on the cluster containing the index.
+
+### Secret
+
+A secret is a new type of database object that securely stores a confidential
+credential.
+
+To create a new secret:
+
+```
+CREATE SECRET <name> AS '<secret>';
+```
+
+To drop an existing secret:
+
+```
+DROP SECRET <name>;
+```
+
+The existence of the secret will be stored in the catalog, but the actual
+contents of the secret will be securely stored in a secrets management system.
+
+Sources and sinks will need to learn to accept secret references in places that
+they currently expect an inline string or file path. Designing that new syntax
+is TODO.
+
+Referencing a secret in a source or sink definition requires that the user who
+creates the source or sink has `USAGE` privilege on the secret.
```

Comment:
We say above that "...if the source references a secret, privileges on that secret are not considered when determining whether a user has access to the source." Aren't these two things contradicting? If not, where am I misunderstanding?

Reply:
Ah, hard to word. I tried to make it a bit clearer:

> Note that if the source references a [secret](#secret), privileges on that
secret are *not* considered when determining whether a user can read the data in
the source. In other words, the `SELECT` privilege on the source grants access
to the data protected by the secret, but not to the secret itself.

Does that help?

Reply:
Yes that makes sense now, thanks!